File: src/main/csharp/Lsp/Lsp/RubberDuck.cs
using System;

namespace Lsp
{
    public class RubberDuck : Duck
    {
        public override void Display()
        {
            Console.WriteLine("This is a rubber duck.");
        }

        public override void Quack()
        {
            squeek();
        }

        public override void Fly()
        {
            // Override to do nothing.
        }

        private void squeek()
        {
            Console.WriteLine("squeek");
        }
    }
}

File: src/main/java/com/methodpark/cmpe/code/lsp/RubberDuck.java
package com.methodpark.cmpe.code.lsp;

public class RubberDuck extends Duck
{
    public void display()
    {
        System.out.println("This is a rubber duck.");
    }

    @Override
    public void quack()
    {
        squeek();
    }

    @Override
    public void fly()
    {
        // Override to do nothing.
    }

    void squeek()
    {
        System.out.println("squeak");
    }

}
